packages/core/src/types.ts

```diff
@@ -2219,3 +2241,53 @@ export type PersistedMachineState<TState extends AnyState> = Pick<
     };
   };
 };
+
+export interface InspectedActorObject {
+  actorRef: AnyActorRef; // local-only
+  sessionId: string;
+  parentId?: string; // Session ID
+  systemId?: string; // Session ID
+  events: ActorCommunicationEvent[]; // events where targetId === sessionId
+  definition?: string; // JSON-stringified machine definition or URL
+  createdAt: number; // Timestamp
+  updatedAt: number; // Timestamp, derived from latest update createdAt
+  status: 0 | 1 | 2; // 0 = not started, 1 = started, 2 = stopped, derived from latest update status
+}
+
+export interface BaseInspectionEvent {
+  actorSystemId: string; // the session ID of the root
+  createdAt: string; // Timestamp
+  id: string; // unique string for this actor update
+}
+
+export interface ActorTransitionEvent {
+  type: '@xstate.snapshot';
+  snapshot: any;
+  event: AnyEventObject; // { type: string, ... }
+  status: 0 | 1 | 2; // 0 = not started, 1 = started, 2 = stopped
+  sessionId: string;
+  actorRef: AnyActorRef; // Only available locally
+}
+
+export interface ActorCommunicationEvent {
+  type: '@xstate.event';
+  event: AnyEventObject; // { type: string, ... }
+  sourceId: string | undefined; // Session ID
+  targetId: string; // Session ID, required
+}
+
+export interface ActorRegistrationEvent {
+  type: '@xstate.actor';
+  actorRef: AnyActorRef;
+  sessionId: string;
+  parentId?: string;
+}
+
+export type InspectionEvent =
+  | ActorTransitionEvent
+  | ActorCommunicationEvent
+  | ActorRegistrationEvent;
```

Comment:
All of those have somewhat confusing names. I think it would be better if the name of the type would match the `.type` closer - but currently, we have `ActorTransitionEvent` with `.type: '@xstate.snapshot'` etc.